roboflow/__init__.py

```diff
@@ -28,11 +29,23 @@ def check_key(api_key, model, notebook):
         else:
             # validate key normally
             response = requests.post(API_URL + "/?api_key=" + api_key)
-            r = response.json()
 
-            if "error" in r or response.status_code != 200:
+            if response.status_code == 401:
                 raise RuntimeError(response.text)
+
+            if response.status_code != 200:
+                # retry 5 times
+                if num_retries < 5:
+                    print("retrying...")
+                    time.sleep(1)
+                    num_retries += 1
+                    return check_key(api_key, model, notebook, num_retries)
+                else:
+                    raise RuntimeError(
+                        "There was an error validating the api key with Roboflow server."
+                    )
             else:
+                r = response.json()
```

Comment:
Should we put a try/catch around this & retry if it fails to produce valid JSON (but wit ha 200 error code) too?

Reply:
We should prob do this retry logic around all API routes not just the initialization

Reply:
Thought about that... I think this is safe if we get a 200 code, it will always produce valid JSON

Reply:
Interesting on _all_ routes - before undertaking that, do we think all api routes have this same vulnerability?